python/tank/template.py

```diff
@@ -259,7 +259,9 @@ def apply_fields(self, fields, platform=None):
         """
         return self._apply_fields(fields, platform=platform)
 
-    def _apply_fields(self, fields, ignore_types=None, platform=None):
+    def _apply_fields(
+        self, fields, ignore_types=None, platform=None, use_defaults=True
```

Comment:
For better or for worse, we should keep the same naming convention of skip_defaults. Your assignment skip_defaults=use_defaults would confuse any readers of this code. Right now it looks like a bug.

Please also document the parameter in the method.

Reply:
The one issue I have with this is that the methods serve different functions. 

- _apply_fields is saying "Fill in these values" so a parameter called "skip_defaults" would make me think "Fill in all values except for the defaults"
- _missing_keys is saying "list any fields that are missing" and "skip_defaults" reads as "but ignore missing default fields"

Because of the way the method is used, in order for the above logic to stay consistent, it would need to be something like:
```python
def _apply_fields(... skip_defaults=False):
    ...
    missing_keys= self._missing_keys(..., skip_defaults=not skip_defaults) 
```
which can also be confusing. 

That's why I decided to change the name, but I'm good with changing it back as well, I just don't think it should be a straight assignment, but a negated assignment

Reply:
Ah, right! Well, then please document the hell out of it when you are making the assignment in the method call. 😄